src/main/scala/util/ClockDivider.scala

```diff
@@ -0,0 +1,41 @@
+package util
+
+import Chisel._
+
+/** Divide the clock by 2 */
+class ClockDivider2 extends Module {
+  val io = new Bundle {
+    val reset_out = Bool(OUTPUT)
+    val clock_out = Clock(OUTPUT)
+  }
+
+  val clock_reg = Reg(Bool())
+  clock_reg := !clock_reg
+
+  // Hold reset_out high until positive edge of clock_out
+  val reset_reg = RegEnable(Bool(false), Bool(true), clock_reg)
+
+  io.clock_out := clock_reg.asClock
+  io.reset_out := reset_reg
+}
+
+class ClockDivider(n: Int) extends Module {
```

Comment:
Why don't you just call this "Power of 2 Clock Divider" and then take po2 as the input. Rather than checking that it is a power of 2 and then taking the log of the number.